packages/substrate-bindings/src/storage.ts

```diff
@@ -33,18 +33,20 @@ export const Storage = (pallet: string) => {
   const palledEncoded = Twox128(textEncoder.encode(pallet))
   return <T, A extends Array<EncoderWithHash<any>>>(
     name: string,
-    dec: Decoder<T>,
+    value: Codec<T>,
     ...encoders: [...A]
   ): {
```

Comment:
it's a bit weird that this function takes `value` just to return it, without actually needing it... specially since now it just returns it alongside another property "keys".

Maybe it would be worth refactoring all of this, I'd propose something like:

```ts
export type KeyCodec<A extends Array<EncoderWithHash<any>> = {
  enc: (
    ...args: {
      [K in keyof A]: A[K] extends EncoderWithHash<infer V> ? V : unknown
    }
  ) => string,
  dec: (value: string) => {
    [K in keyof A]: A[K] extends EncoderWithHash<infer V> ? V : unknown
  }
}
export const StorageKeys = (pallet: string) =>
  <A extends Array<…>>(name: string, ...encoders: […A]): KeyCodec<A> => { … }

// …
const storageWithFallback = (
  len: number,
  name: string,
  value: Codec<any>,
  ...args: Array<EncoderWithHash<any>>
) => {
  const keys = storagePallet!(name, ...args)
  return {
    keys,
    value,
    len,
    fallback:
      storageEntry.modifier === 1
        ? result.dec(storageEntry.fallback)
        : undefined,
  }
}
```